test-data/unit/check-unreachable-code.test

```diff
@@ -302,6 +302,8 @@ else:
 import sys
 if sys.platform == 'fictional':
     def foo() -> int: return 0
+if platform.system() == 'fake':
+    def foo() -> int: return 0
```

Comment:
There should also be a test that has a successful platform check.

Also test that `--platform` affects a `platform.system()` check.